deploy/deployUtils.ts

```diff
@@ -765,3 +770,166 @@ export async function stealFundsFromWhales(
+      log: true,
+      skipIfAlreadyDeployed: true,
+      args: [masterRegistryAddress],
+      libraries: {
+        SwapUtils: swapUtilsAddress,
+        AmplificationUtils: amplificationUtilsAddress,
+      },
+    })
+    console.log(
+      `PermissionlessSwap deployed at ${permissionlessSwapDeployment.address}`,
+    )
+  }
+  const permissionlessSwapAddress = (await get("PermissionlessSwap")).address
+
+  // deploy PermissionlessMetaSwap if needed
+  if (permissionlessMetaSwap == null) {
+    console.log("PermissionlessMetaSwap not found, deploying")
+    const permissionlessMetaSwapDeployment = await deploy(
+      "PermissionlessMetaSwap",
+      {
+        from: deployer,
+        log: true,
+        skipIfAlreadyDeployed: true,
+        args: [masterRegistryAddress],
+        libraries: {
+          SwapUtils: swapUtilsAddress,
+          MetaSwapUtils: (await get("MetaSwapUtils")).address,
+          AmplificationUtils: amplificationUtilsAddress,
+        },
+      },
+    )
+    console.log(
+      `PermissionlessMetaSwap deployed at ${permissionlessMetaSwapDeployment.address}`,
+    )
+  }
+  const permissionlessMetaSwapAddress = (await get("PermissionlessMetaSwap"))
+    .address
+
+  // deploy PermissionlessDeployer if needed
+  if (permissionlessDeployer == null) {
+    console.log("PermissionlessDeployer not found, deploying")
+    const PermissionlessDeployerDeployment = await deploy(
+      "PermissionlessDeployer",
+      {
+        from: deployer,
+        log: true,
+        skipIfAlreadyDeployed: true,
+        args: [
+          deployer,
+          masterRegistryAddress,
+          (await get("LPToken")).address,
+          permissionlessSwapAddress,
+          permissionlessMetaSwapAddress,
+          // Below needs to be a non-clone instance of the MetaswapDeposit Contract
+          metaswapDeposit,
+        ],
+      },
+    )
+    console.log(
+      `Deployed PermissionlessDeployer at ${PermissionlessDeployerDeployment.address}`,
+    )
+
+    // set target swaps for the permissionless deployer,
+    // PermissionlessDeployer to the master registry
+    console.log("Setting target swaps for PermissionlessDeployer")
+    console.log(`PermissionlessSwap at: ${permissionlessSwapAddress}`)
+    console.log(`PermissionlessMetaSwap at: ${permissionlessMetaSwapAddress}`)
+    await execute(
+      "PermissionlessDeployer",
+      { from: deployer, log: true },
+      "setTargetSwap",
+      permissionlessSwapAddress,
+    )
+
+    await execute(
+      "PermissionlessDeployer",
+      { from: deployer, log: true },
+      "setTargetMetaSwap",
+      permissionlessMetaSwapAddress,
+    )
+
+    console.log("Adding PermissionlessDeployer to MasterRegistry")
+    await execute(
+      "MasterRegistry",
+      {
+        from: deployer,
+        log: true,
+      },
+      "addRegistry",
+      ethers.utils.formatBytes32String("PermissionlessDeployer"),
+      PermissionlessDeployerDeployment.address,
+    )
```

Comment:
Is it necessary to set storage variables again when you already passed the info in the constructor?